fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/FDBLuceneQueryTest.java

```diff
@@ -1271,4 +1277,55 @@ void continuations(boolean sorted) throws Exception {
             }
         }
     }
+
+    /**
+     * This tests is reproducing a thread deadlock that happened with Lucene threads when reading schema information.
+     * The fix for the deadlock has removed the {@code Map.computeIfAbsent} call (with the {@code synchronized} block)
+     * that blocked the {@code asyncToSync} calls.
+     * This test creates a large enough set of {@link org.apache.lucene.index.SegmentReader} futures so as to flood the thread pool,
+     * and then queries Lucene. THe large number of futures in the pool, together with the {@code synchronized} block
+     * would cause a deadlock.
+     */
+    @Test
+    void testQueryWithManyDocuments() {
+        String[] words = generateRandomWords(1000);
```

Comment:
It's probably worth commenting here that we generate random words, rather than using the `TextSamples`, or choosing random words from the `TextSamples`, because Lucene is too efficient at compressing those, so even with a lot of documents (100k+ If I remember correctly), it still falls short of creating enough segments to cause a problem.

Reply:
Done.